src/graph/optimizer/rule/PushFilterDownAppendVerticesRule.cpp

```diff
@@ -0,0 +1,118 @@
+/* Copyright (c) 2023 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#include "graph/optimizer/rule/PushFilterDownAppendVerticesRule.h"
+
+#include "common/expression/Expression.h"
+#include "graph/optimizer/OptContext.h"
+#include "graph/optimizer/OptGroup.h"
+#include "graph/planner/plan/PlanNode.h"
+#include "graph/planner/plan/Query.h"
+#include "graph/util/ExpressionUtils.h"
+#include "graph/visitor/ExtractFilterExprVisitor.h"
+
+using nebula::Expression;
+using nebula::graph::AppendVertices;
+using nebula::graph::Filter;
+using nebula::graph::PlanNode;
+using nebula::graph::QueryContext;
+
+namespace nebula {
+namespace opt {
+
+std::unique_ptr<OptRule> PushFilterDownAppendVerticesRule::kInstance =
+    std::unique_ptr<PushFilterDownAppendVerticesRule>(new PushFilterDownAppendVerticesRule());
+
+PushFilterDownAppendVerticesRule::PushFilterDownAppendVerticesRule() {
+  RuleSet::QueryRules().addRule(this);
+}
+
+const Pattern &PushFilterDownAppendVerticesRule::pattern() const {
+  static Pattern pattern =
+      Pattern::create(PlanNode::Kind::kFilter, {Pattern::create(PlanNode::Kind::kAppendVertices)});
+  return pattern;
+}
+
+StatusOr<OptRule::TransformResult> PushFilterDownAppendVerticesRule::transform(
+    OptContext *ctx, const MatchedResult &matched) const {
+  auto filterGroupNode = matched.node;
+  auto appendVerticesGroupNode = matched.dependencies.front().node;
+  auto filter = static_cast<const Filter *>(filterGroupNode->node());
+  auto appendVertices = static_cast<const AppendVertices *>(appendVerticesGroupNode->node());
+  auto qctx = ctx->qctx();
+  auto pool = qctx->objPool();
+  auto condition = graph::ExpressionUtils::rewriteVertexPropertyFilter(
+      pool, appendVertices->colNames().back(), filter->condition()->clone());
+
```

Comment:
Need to port the latest version of this file from the ent repo. I introduced a bug fix here at line:48.

Reply:
Which pr?